portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -50,13 +52,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty("name", name);
     commentEntity.setProperty("text", text);
     commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("numReports", 0);
 
     datastore.put(commentEntity);
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comment");
+    query.addSort("numReports", SortDirection.ASCENDING); // required sorting to apply filter
+    query.addSort("timestamp", SortDirection.DESCENDING); // display in order of recency
+    query = query.addFilter("numReports", Query.FilterOperator.LESS_THAN, 2); // only show comments with less than 2 reports
```

Comment:
Comments are redundant because the code is self explanatory. Check out the `To Comment or Not to Comment?` ToTT.